### PR TITLE
⭐ github.packages resource

### PR DESCRIPTION
**Pull request by slntopp**

### CODE REVIEW

File: providers/github/resources/github_package.go [hunk 1]
```diff
@@ -12,3 +22,96 @@ func (g *mqlGithubPackage) id() (string, error) {
...
+func (g *mqlGithubPackages) id() (string, error) {
+	return "github.packages", nil
+}
+
+func (g *mqlGithubPackages) list() ([]any, error) {
+	return newMqlGithubPackages(g.MqlRuntime, nil)
+}
+
+func (g *mqlGithubPackages) public() ([]any, error) {
+	visibility := "public"
```
Comment (czunker) [inline providers/github/resources/github_package.go]: nit: Perhaps use some `const` for different visibilities?

File: providers/github/resources/github_package.go [hunk 2]
```diff
@@ -12,3 +22,96 @@ func (g *mqlGithubPackage) id() (string, error) {
...
+}
+
+func newMqlGithubPackages(runtime *plugin.Runtime, visibility *string) ([]any, error) {
+	conn := runtime.Connection.(*connection.GithubConnection)
+	orgId, err := conn.Organization()
+	if err != nil {
+		return nil, err
+	}
+
+	pkgTypes := []string{"npm", "maven", "rubygems", "docker", "nuget", "container"}
```
Comment (czunker) [inline providers/github/resources/github_package.go:62]: Is this what we are interested in or what GitHub provides?

File: providers/github/resources/github_package.go [hunk 3]
```diff
@@ -12,3 +22,96 @@ func (g *mqlGithubPackage) id() (string, error) {
...
+	for i := range pkgTypes {
+		listOpts := &github.PackageListOptions{
+			ListOptions: github.ListOptions{PerPage: paginationPerPage},
+			PackageType: github.String(pkgTypes[i]),
+			Visibility:  visibility,
+		}
+
+		var allPackages []*github.Package
+		for {
+			packages, resp, err := conn.Client().Organizations.ListPackages(context.Background(), orgId.Name, listOpts)
```
Comment (czunker) [inline providers/github/resources/github_package.go:73]: I don't know how this is intended to be used.
But this line makes the output of this command a bit confusing to me:
```
cnquery run github repo mondoohq/packer-plugin-cnspec -c "github.packages"
```
I'd expected only the packages of the specified repo, but I got all across the org. Is this how it should be?

File: providers/github/resources/github_package.go (hunk 2 above)
Comment (slntopp) [reply providers/github/resources/github_package.go:62]: This is all kinds of packages GitHub provides

File: providers/github/resources/github_package.go (hunk 3 above)
Comment (slntopp) [reply providers/github/resources/github_package.go:73]: Packages are not linked to repos, packages are always stored under account/org

So scan still would be with `github org some_org`, just it won't fetch the org

File: providers/github/resources/github_package.go (hunk 1 above)
Comment (slntopp) [reply providers/github/resources/github_package.go]: defined `const`s

**Review by imilchev — APPROVED** [review]
LGTM

File: providers/github/resources/github_package.go (hunk 3 above)
Comment (czunker) [reply providers/github/resources/github_package.go:73]: Ah, thanks for the background.

**Review by czunker — APPROVED** [review]
Thanks @slntopp

**Pull request merged**